frontend/src/components/nodes/ValidatorsList.tsx

```diff
@@ -21,11 +21,36 @@ class ValidatorsList extends Component<Props> {
 
     let validatorsList = validators.sort(
       (a: N.ValidationNodeInfo, b: N.ValidationNodeInfo) => {
-        return new BN(b.currentStake).sub(new BN(a.currentStake));
+        // we take "active", "joining", "leaving" validators and sort them firstly
+        // after then we sort the rest
+        const validatingGroup = ["active", "joining", "leaving"];
+        if (
+          a.stakingStatus &&
+          validatingGroup.indexOf(a.stakingStatus) >= 0 &&
+          b.stakingStatus &&
+          validatingGroup.indexOf(b.stakingStatus) >= 0
+        ) {
+          return new BN(b.currentStake).cmp(new BN(a.currentStake));
+        } else if (
+          (a.proposedStake || b.proposedStake) &&
+          a.stakingStatus &&
+          validatingGroup.indexOf(a.stakingStatus) < 0 &&
+          b.stakingStatus &&
+          validatingGroup.indexOf(b.stakingStatus) < 0
+        ) {
+          return new BN(b.proposedStake || 0).cmp(new BN(a.proposedStake || 0));
+        } else if (
+          a.stakingStatus &&
+          validatingGroup.indexOf(a.stakingStatus) < 0 &&
+          b.stakingStatus &&
+          validatingGroup.indexOf(b.stakingStatus) < 0
+        ) {
+          return new BN(b.currentStake).cmp(new BN(a.currentStake));
+        }
```

Comment:
How do you feel about this sorting? (I am not sure if `currentStake` is always defined, so I added `|| 0`)

```suggestion
        const aInValidatingGroup = validatingGroup.indexOf(a.stakingStatus) >= 0;
        const bInValidatingGroup = validatingGroup.indexOf(b.stakingStatus) >= 0;
        if (aInValidatingGroup && bInValidatingGroup) {
          return new BN(b.currentStake).cmp(new BN(a.currentStake));
        } else {
          const aStake = BN.max(new BN(b.currentStake || 0), new BN(b.proposedStake || 0));
          const bStake = BN.max(new BN(a.currentStake || 0), new BN(a.proposedStake || 0));
          return aStake.cmp(bStake);
        }
```